src/types/vault.ts

```diff
@@ -28,6 +34,114 @@ export interface VaultExt<WrappedUnit extends BitcoinUnit> {
+        this.liquidatedCollateral = liquidatedCollateral;
+        this.collateralCurrency = collateralCurrency;
+    }
+
+    getRedeemableTokens(): MonetaryAmount<Currency<WrappedUnit>, WrappedUnit> {
+        return this.issuedTokens.sub(this.toBeRedeemedTokens);
+    }
+
+    async getIssuableTokens(): Promise<MonetaryAmount<Currency<WrappedUnit>, WrappedUnit>> {
+        const isBanned = await this.isBanned();
+        if (isBanned) {
+            return newMonetaryAmount(0, currencyIdToMonetaryCurrency(this.id.currencies.wrapped));
+        }
+        const freeCollateral = await this.getFreeCollateral();
+        const secureCollateralThreshold = await this.getSecureCollateralThreshold();
+        const oracleAPI = new DefaultOracleAPI(this.api, currencyIdToMonetaryCurrency(this.id.currencies.wrapped));
+        const backableWrappedTokens = await oracleAPI.convertCollateralToWrapped(freeCollateral);
+        // Force type-assert here as the oracle API only uses wrapped Bitcoin
+        return backableWrappedTokens.div(secureCollateralThreshold) as unknown as MonetaryAmount<Currency<WrappedUnit>, WrappedUnit>;
+    }
+
+    async isBanned(): Promise<boolean> {
+        const systemAPI = new DefaultSystemAPI(this.api);
+        const currentBlockNumber = await systemAPI.getCurrentActiveBlockNumber();
+        if (!this.bannedUntil) {
+            return false;
+        }
+        return this.bannedUntil >= currentBlockNumber;
+    }
+
+    getBackedTokens(): MonetaryAmount<Currency<WrappedUnit>, WrappedUnit> {
+        return this.issuedTokens.add(this.toBeIssuedTokens);
+    }
+
+    async getFreeCollateral(): Promise<MonetaryAmount<Currency<CollateralUnit>, CollateralUnit>> {
+        const usedCollateral = await this.getUsedCollateral();
+        const totalCollateral = await this.computeBackingCollateral();
+        return totalCollateral.sub(usedCollateral);
+    }
+
+    async getUsedCollateral(): Promise<MonetaryAmount<Currency<CollateralUnit>, CollateralUnit>> {
+        const oracleAPI = new DefaultOracleAPI(this.api, currencyIdToMonetaryCurrency(this.id.currencies.wrapped));
+        const backedTokens = this.getBackedTokens();
+        const backedTokensInCollateral = await oracleAPI.convertWrappedToCollateral(
+            // Force type-assert here as the oracle API only uses wrapped Bitcoin
+            backedTokens as unknown as MonetaryAmount<Currency<BitcoinUnit>, BitcoinUnit>,
+            currencyIdToMonetaryCurrency(this.id.currencies.collateral) as Currency<CollateralUnit>
+        );
+        const secureCollateralThreshold = await this.getSecureCollateralThreshold();
+        const usedCollateral = backedTokensInCollateral.mul(secureCollateralThreshold);
+        const totalCollateral = await this.computeBackingCollateral();
+        return totalCollateral.min(usedCollateral);
+    }
+
+    // TODO: The functions below are duplicated from other APIs. Remove them after APIs
```

Comment:
Is this something that you want to do in this PR or later?

Reply:
It's for later